src/shogun/lib/any.h

```diff
@@ -1274,15 +1290,30 @@ namespace shogun
 	inline void register_casts()
 	{
 		using Derived = std::remove_pointer_t<T>;
-		if constexpr (std::is_base_of_v<CSGObject, Derived>)
+		if constexpr (std::is_base_of_v<SGObject, Derived>)
```

Comment:
@lisitsyn do we wanna have support for SGO* ? as you can see the same thing is copypasted below for shared_ptr<SGO-like>